gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -1065,13 +1065,14 @@ public void run() {
             callbacks.getStoragePoolSize(true);
       }
       this.memorySize = memorySize;
-      if (memorySize > 0) {
+      if (memorySize > 0 && CallbackFactoryProvider.getStoreCallbacks().isEnterpriseEdition()) {
```

Comment:
GemFireVersion is a singleton which is set with whatever it is setup with at boot. The code you added in SnappySession.isEnterpriseEdition itself also uses that:

```scala
    GemFireVersion.getInstance(classOf[SnappyDataVersion],
      SharedUtils.GFXD_VERSION_PROPERTIES)
    GemFireVersion.isEnterpriseEdition
```
I had checked that just GemFireVersion.isEnterpriseEdition works just fine in my local changes.